pkg/generate/app/cmd/newapp.go

```diff
@@ -219,30 +221,80 @@ func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespac
 	}
 }
 
+func (c *AppConfig) addEnvironmentArguments(s string) bool {
+	rc := cmdutil.IsEnvironmentArgument(s)
+	if rc {
+		glog.V(2).Infof("treating %s as possible environment argument\n", s)
+		c.Environment = append(c.Environment, s)
+	}
+	return rc
+}
+
+func (c *AppConfig) addSourceArguments(s string) bool {
+	remote, rerr := app.IsRemoteRepository(s)
+	local, derr := app.IsDirectory(s)
+
+	if remote || local {
+		glog.V(2).Infof("treating %s as possible source repo\n", s)
+		c.SourceRepositories = append(c.SourceRepositories, s)
+		return true
+	}
+
+	if rerr != nil {
+		c.ArgumentClassificationErrors["Remote access"] = rerr
+	}
+
+	if derr != nil {
+		c.ArgumentClassificationErrors["Local access"] = derr
+	}
+
+	return false
+}
+
+func (c *AppConfig) addComponentArguments(s string) bool {
+	err := app.IsComponentReference(s)
+	if err == nil {
+		glog.V(2).Infof("treating %s as a component ref\n", s)
+		c.Components = append(c.Components, s)
+		return true
+	}
+	c.ArgumentClassificationErrors["Component reference processing"] = err
+
+	return false
+}
+
+func (c *AppConfig) addTemplateArguments(s string) bool {
+	rc, err := app.IsPossibleTemplateFile(s)
+	if rc {
+		glog.V(2).Infof("treating %s as possible template file\n", s)
+		c.Components = append(c.Components, s)
+		return true
+	}
+	if err != nil {
+		c.ArgumentClassificationErrors["Template processing"] = err
+	}
+	return false
+}
+
 // AddArguments converts command line arguments into the appropriate bucket based on what they look like
 func (c *AppConfig) AddArguments(args []string) []string {
 	unknown := []string{}
+	c.ArgumentClassificationErrors = map[string]error{}
 	for _, s := range args {
+		if len(s) == 0 {
+			continue
+		}
+
 		switch {
-		case cmdutil.IsEnvironmentArgument(s):
-			glog.V(2).Infof("treating %s as possible environment argument\n", s)
-			c.Environment = append(c.Environment, s)
-		case app.IsPossibleSourceRepository(s):
-			glog.V(2).Infof("treating %s as possible source repo\n", s)
-			c.SourceRepositories = append(c.SourceRepositories, s)
-		case app.IsComponentReference(s):
-			glog.V(2).Infof("treating %s as a component ref\n", s)
-			c.Components = append(c.Components, s)
-		case app.IsPossibleTemplateFile(s):
-			glog.V(2).Infof("treating %s as possible template file\n", s)
-			c.Components = append(c.Components, s)
+		case c.addEnvironmentArguments(s):
+		case c.addSourceArguments(s):
+		case c.addComponentArguments(s):
+		case c.addTemplateArguments(s):
```

Comment:
if the user passes multiple arguments, this error accumulation and reporting logic breaks down.  You need to accumulate the ArgumentClassificationErrors on a per argument basis.